bin/git-co-upstream-pr

```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+set -e
+
+pull_request_number=$1
+local_branch_name=$2
+
+if git remote -v | grep upstream; then
+  git fetch upstream "pull/$pull_request_number/head:$local_branch_name"
+  git checkout "$local_branch_name"
+else
+  echo "You don't have an upstream remote set.
+  Use:
+  git remote add upstream {upstream_remote_url}
```

Comment:
Is `upstream` different from `origin`? Maybe I don't know how to use this command.

Reply:
Yes, they are different! `upstream` repositories are the original repository that a fork is created from. GitHub has some details here: https://docs.github.com/en/get-started/quickstart/fork-a-repo#configuring-git-to-sync-your-fork-with-the-upstream-repository

I have mainly used `upstream` remote repos for forked open source projects. But they can be used for any repos where you want to have your own fork, but still wants to fetch the original repo's updates.